### ISSUE TITLE
Cross compiling for armv5te.

### ISSUE THREAD

**Issue by mosolovsa** [japaric/xargo#163]
Hi! Thanks for the progress you've already done to make rust cross compilation simple.

I need some help with cross compiling for armv5te (ARM926EJ-S CPU).
I have an issue, that must be somehow related to: 
[No atomic instructions on ARMv5](https://github.com/rust-lang/rust/blob/master/src/librustc_back/target/armv5te_unknown_linux_gnueabi.rs#L30)
In the result of this alloc::arc::Arc is not presented in binary.

When I try to use xargo:
```
xargo build --release --target=armv5te-unknown-linux-gnueabi
```
with Xargo.toml
```
[dependencies]
std = {default-features=false, features = ["jemalloc"]}
```
I've got the:
<details> 
  <summary>long console output complaining on errors</summary>
   Updating registry `https://github.com/rust-lang/crates.io-index`
   Compiling libc v0.2.29
   Compiling core v0.0.0 (file:///home/user/.rustup/toolchains/nightly-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/src/libcore)
   Compiling libc v0.0.0 (file:///home/user/.rustup/toolchains/nightly-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/src/rustc/libc_shim)
   Compiling gcc v0.3.53
   Compiling unwind v0.0.0 (file:///home/user/.rustup/toolchains/nightly-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/src/libunwind)
   Compiling filetime v0.1.10
   Compiling build_helper v0.1.0 (file:///home/user/.rustup/toolchains/nightly-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/src/build_helper)
   Compiling alloc_jemalloc v0.0.0 (file:///home/user/.rustup/toolchains/nightly-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/src/liballoc_jemalloc)
   Compiling compiler_builtins v0.0.0 (file:///home/user/.rustup/toolchains/nightly-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/src/rustc/compiler_builtins_shim)
   Compiling std v0.0.0 (file:///home/user/.rustup/toolchains/nightly-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/src/libstd)
warning: use of deprecated item: gcc::Config has been renamed to gcc::Build
    --> /home/user/.rustup/toolchains/nightly-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/src/rustc/compiler_builtins_shim/../../libcompiler_builtins/build.rs:4011:24
     |
4011 |         let cfg = &mut gcc::Config::new();
     |                        ^^^^^^^^^^^^^^^^
     |
     = note: #[warn(deprecated)] on by default

warning: ../../libcompiler_builtins/compiler-rt/lib/builtins/divdc3.c:21:1: warning: conflicting types for built-in function '__divdc3' [enabled by default]
warning:  __divdc3(double __a, double __b, double __c, double __d)
warning:  ^
warning: ../../libcompiler_builtins/compiler-rt/lib/builtins/divsc3.c:21:1: warning: conflicting types for built-in function '__divsc3' [enabled by default]
warning:  __divsc3(float __a, float __b, float __c, float __d)
warning:  ^
warning: ../../libcompiler_builtins/compiler-rt/lib/builtins/int_util.c: In function 'compilerrt_abort_impl':
warning: ../../libcompiler_builtins/compiler-rt/lib/builtins/int_util.c:57:40: warning: unused parameter 'file' [-Wunused-parameter]
warning:  void compilerrt_abort_impl(const char *file, int line, const char *function) {
warning:                                         ^
warning: ../../libcompiler_builtins/compiler-rt/lib/builtins/int_util.c:57:50: warning: unused parameter 'line' [-Wunused-parameter]
warning:  void compilerrt_abort_impl(const char *file, int line, const char *function) {
warning:                                                   ^
warning: ../../libcompiler_builtins/compiler-rt/lib/builtins/int_util.c:57:68: warning: unused parameter 'function' [-Wunused-parameter]
warning:  void compilerrt_abort_impl(const char *file, int line, const char *function) {
warning:                                                                     ^
warning: ../../libcompiler_builtins/compiler-rt/lib/builtins/muldc3.c:21:1: warning: conflicting types for built-in function '__muldc3' [enabled by default]
warning:  __muldc3(double __a, double __b, double __c, double __d)
warning:  ^
warning: ../../libcompiler_builtins/compiler-rt/lib/builtins/mulsc3.c:21:1: warning: conflicting types for built-in function '__mulsc3' [enabled by default]
warning:  __mulsc3(float __a, float __b, float __c, float __d)
warning:  ^
   Compiling std_unicode v0.0.0 (file:///home/user/.rustup/toolchains/nightly-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/src/libstd_unicode)
   Compiling rand v0.0.0 (file:///home/user/.rustup/toolchains/nightly-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/src/librand)
   Compiling panic_abort v0.0.0 (file:///home/user/.rustup/toolchains/nightly-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/src/libpanic_abort)
   Compiling alloc v0.0.0 (file:///home/user/.rustup/toolchains/nightly-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/src/liballoc)
   Compiling collections v0.0.0 (file:///home/user/.rustup/toolchains/nightly-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/src/libcollections)
   Compiling alloc_system v0.0.0 (file:///home/user/.rustup/toolchains/nightly-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/src/liballoc_system)
error[E0432]: unresolved import `alloc::arc::Arc`
  --> /home/user/.rustup/toolchains/nightly-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/src/libstd/sync/mod.rs:21:22
   |
21 | pub use alloc::arc::{Arc, Weak};
   |                      ^^^ Could not find `arc` in `alloc`

error[E0432]: unresolved import `alloc::arc::Weak`
  --> /home/user/.rustup/toolchains/nightly-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/src/libstd/sync/mod.rs:21:27
   |
21 | pub use alloc::arc::{Arc, Weak};
   |                           ^^^^ Could not find `arc` in `alloc`

error[E0432]: unresolved import `sync::atomic::AtomicBool`
  --> /home/user/.rustup/toolchains/nightly-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/src/libstd/sync/mpsc/blocking.rs:14:20
   |
14 | use sync::atomic::{AtomicBool, Ordering};
   |                    ^^^^^^^^^^ no `AtomicBool` in `sync::atomic`

error[E0432]: unresolved import `sync::atomic::AtomicUsize`
  --> /home/user/.rustup/toolchains/nightly-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/src/libstd/sync/mpsc/oneshot.rs:44:20
   |
44 | use sync::atomic::{AtomicUsize, Ordering};
   |                    ^^^^^^^^^^^ no `AtomicUsize` in `sync::atomic`

error[E0432]: unresolved import `sync::atomic::AtomicUsize`
  --> /home/user/.rustup/toolchains/nightly-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/src/libstd/sync/mpsc/shared.rs:29:20
   |
29 | use sync::atomic::{AtomicUsize, AtomicIsize, AtomicBool, Ordering};
   |                    ^^^^^^^^^^^ no `AtomicUsize` in `sync::atomic`

error[E0432]: unresolved import `sync::atomic::AtomicIsize`
  --> /home/user/.rustup/toolchains/nightly-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/src/libstd/sync/mpsc/shared.rs:29:33
   |
29 | use sync::atomic::{AtomicUsize, AtomicIsize, AtomicBool, Ordering};
   |                                 ^^^^^^^^^^^ no `AtomicIsize` in `sync::atomic`

error[E0432]: unresolved import `sync::atomic::AtomicBool`
  --> /home/user/.rustup/toolchains/nightly-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/src/libstd/sync/mpsc/shared.rs:29:46
   |
29 | use sync::atomic::{AtomicUsize, AtomicIsize, AtomicBool, Ordering};
   |                                              ^^^^^^^^^^ no `AtomicBool` in `sync::atomic`

error[E0432]: unresolved import `sync::atomic::AtomicIsize`
  --> /home/user/.rustup/toolchains/nightly-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/src/libstd/sync/mpsc/stream.rs:32:20
   |
32 | use sync::atomic::{AtomicIsize, AtomicUsize, Ordering, AtomicBool};
   |                    ^^^^^^^^^^^ no `AtomicIsize` in `sync::atomic`

error[E0432]: unresolved import `sync::atomic::AtomicUsize`
  --> /home/user/.rustup/toolchains/nightly-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/src/libstd/sync/mpsc/stream.rs:32:33
   |
32 | use sync::atomic::{AtomicIsize, AtomicUsize, Ordering, AtomicBool};
   |                                 ^^^^^^^^^^^ no `AtomicUsize` in `sync::atomic`

error[E0432]: unresolved import `sync::atomic::AtomicBool`
  --> /home/user/.rustup/toolchains/nightly-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/src/libstd/sync/mpsc/stream.rs:32:56
   |
32 | use sync::atomic::{AtomicIsize, AtomicUsize, Ordering, AtomicBool};
   |                                                        ^^^^^^^^^^ no `AtomicBool` in `sync::atomic`

error[E0432]: unresolved import `sync::atomic::AtomicUsize`
  --> /home/user/.rustup/toolchains/nightly-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/src/libstd/sync/mpsc/sync.rs:44:30
   |
44 | use sync::atomic::{Ordering, AtomicUsize};
   |                              ^^^^^^^^^^^ no `AtomicUsize` in `sync::atomic`

error[E0432]: unresolved import `sync::atomic::AtomicPtr`
  --> /home/user/.rustup/toolchains/nightly-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/src/libstd/sync/mpsc/mpsc_queue.rs:30:20
   |
30 | use sync::atomic::{AtomicPtr, Ordering};
   |                    ^^^^^^^^^ no `AtomicPtr` in `sync::atomic`. Did you mean to use `atomic_or`?

error[E0432]: unresolved import `sync::atomic::AtomicPtr`
  --> /home/user/.rustup/toolchains/nightly-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/src/libstd/sync/mpsc/spsc_queue.rs:23:20
   |
23 | use sync::atomic::{AtomicPtr, AtomicUsize, Ordering};
   |                    ^^^^^^^^^ no `AtomicPtr` in `sync::atomic`. Did you mean to use `atomic_or`?

error[E0432]: unresolved import `sync::atomic::AtomicUsize`
  --> /home/user/.rustup/toolchains/nightly-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/src/libstd/sync/mpsc/spsc_queue.rs:23:31
   |
23 | use sync::atomic::{AtomicPtr, AtomicUsize, Ordering};
   |                               ^^^^^^^^^^^ no `AtomicUsize` in `sync::atomic`

error[E0432]: unresolved import `sync::atomic::AtomicUsize`
  --> /home/user/.rustup/toolchains/nightly-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/src/libstd/sync/condvar.rs:12:20
   |
12 | use sync::atomic::{AtomicUsize, Ordering};
   |                    ^^^^^^^^^^^ no `AtomicUsize` in `sync::atomic`

error[E0432]: unresolved import `sync::atomic::AtomicUsize`
  --> /home/user/.rustup/toolchains/nightly-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/src/libstd/sync/once.rs:70:20
   |
70 | use sync::atomic::{AtomicUsize, AtomicBool, Ordering};
   |                    ^^^^^^^^^^^ no `AtomicUsize` in `sync::atomic`

error[E0432]: unresolved import `sync::atomic::AtomicBool`
  --> /home/user/.rustup/toolchains/nightly-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/src/libstd/sync/once.rs:70:33
   |
70 | use sync::atomic::{AtomicUsize, AtomicBool, Ordering};
   |                                 ^^^^^^^^^^ no `AtomicBool` in `sync::atomic`

error[E0432]: unresolved import `sync::atomic::AtomicBool`
  --> /home/user/.rustup/toolchains/nightly-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/src/libstd/sys_common/poison.rs:13:20
   |
13 | use sync::atomic::{AtomicBool, Ordering};
   |                    ^^^^^^^^^^ no `AtomicBool` in `sync::atomic`

error[E0432]: unresolved import `sync::atomic::AtomicUsize`
  --> /home/user/.rustup/toolchains/nightly-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/src/libstd/sys_common/thread_local.rs:62:26
   |
62 | use sync::atomic::{self, AtomicUsize, Ordering};
   |                          ^^^^^^^^^^^ no `AtomicUsize` in `sync::atomic`

error[E0432]: unresolved import `sync::atomic::AtomicUsize`
  --> /home/user/.rustup/toolchains/nightly-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/src/libstd/sys/unix/weak.rs:34:20
   |
34 | use sync::atomic::{AtomicUsize, Ordering};
   |                    ^^^^^^^^^^^ no `AtomicUsize` in `sync::atomic`

error[E0432]: unresolved import `sync::atomic::AtomicBool`
  --> /home/user/.rustup/toolchains/nightly-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/src/libstd/sys/unix/fd.rs:17:20
   |
17 | use sync::atomic::{AtomicBool, Ordering};
   |                    ^^^^^^^^^^ no `AtomicBool` in `sync::atomic`

error[E0432]: unresolved import `sync::atomic::AtomicBool`
  --> /home/user/.rustup/toolchains/nightly-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/src/libstd/sys/unix/pipe.rs:14:20
   |
14 | use sync::atomic::{AtomicBool, ATOMIC_BOOL_INIT, Ordering};
   |                    ^^^^^^^^^^ no `AtomicBool` in `sync::atomic`

error[E0432]: unresolved import `sync::atomic::ATOMIC_BOOL_INIT`
  --> /home/user/.rustup/toolchains/nightly-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/src/libstd/sys/unix/pipe.rs:14:32
   |
14 | use sync::atomic::{AtomicBool, ATOMIC_BOOL_INIT, Ordering};
   |                                ^^^^^^^^^^^^^^^^ no `ATOMIC_BOOL_INIT` in `sync::atomic`

error[E0432]: unresolved import `sync::atomic::AtomicBool`
   --> /home/user/.rustup/toolchains/nightly-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/src/libstd/sys/unix/rand.rs:121:28
    |
121 |         use sync::atomic::{AtomicBool, Ordering};
    |                            ^^^^^^^^^^ no `AtomicBool` in `sync::atomic`

error[E0432]: unresolved import `sync::atomic::AtomicUsize`
  --> /home/user/.rustup/toolchains/nightly-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/src/libstd/sys/unix/rwlock.rs:13:20
   |
13 | use sync::atomic::{AtomicUsize, Ordering};
   |                    ^^^^^^^^^^^ no `AtomicUsize` in `sync::atomic`

error[E0412]: cannot find type `Arc` in this scope
   --> /home/user/.rustup/toolchains/nightly-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/src/libstd/thread/mod.rs:352:25
    |
352 |         let my_packet : Arc<UnsafeCell<Option<Result<T>>>>
    |                         ^^^ not found in this scope

error[E0433]: failed to resolve. Use of undeclared type or module `Arc`
   --> /home/user/.rustup/toolchains/nightly-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/src/libstd/thread/mod.rs:353:15
    |
353 |             = Arc::new(UnsafeCell::new(None));
    |               ^^^^^^^^ Use of undeclared type or module `Arc`

error[E0412]: cannot find type `Arc` in this scope
   --> /home/user/.rustup/toolchains/nightly-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/src/libstd/thread/mod.rs:902:12
    |
902 |     inner: Arc<Inner>,
    |            ^^^ not found in this scope

error[E0433]: failed to resolve. Use of undeclared type or module `Arc`
   --> /home/user/.rustup/toolchains/nightly-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/src/libstd/thread/mod.rs:912:20
    |
912 |             inner: Arc::new(Inner {
    |                    ^^^^^^^^ Use of undeclared type or module `Arc`

error[E0412]: cannot find type `Arc` in this scope
    --> /home/user/.rustup/toolchains/nightly-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/src/libstd/thread/mod.rs:1074:18
     |
1074 | struct Packet<T>(Arc<UnsafeCell<Option<Result<T>>>>);
     |                  ^^^ not found in this scope

error[E0412]: cannot find type `Arc` in this scope
  --> /home/user/.rustup/toolchains/nightly-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/src/libstd/io/lazy.rs:19:20
   |
19 |     ptr: Cell<*mut Arc<T>>,
   |                    ^^^ not found in this scope

error[E0412]: cannot find type `Arc` in this scope
  --> /home/user/.rustup/toolchains/nightly-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/src/libstd/io/lazy.rs:20:19
   |
20 |     init: fn() -> Arc<T>,
   |                   ^^^ not found in this scope

error[E0412]: cannot find type `Arc` in this scope
  --> /home/user/.rustup/toolchains/nightly-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/src/libstd/io/lazy.rs:26:36
   |
26 |     pub const fn new(init: fn() -> Arc<T>) -> Lazy<T> {
   |                                    ^^^ not found in this scope

error[E0412]: cannot find type `Arc` in this scope
  --> /home/user/.rustup/toolchains/nightly-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/src/libstd/io/lazy.rs:34:41
   |
34 |     pub fn get(&'static self) -> Option<Arc<T>> {
   |                                         ^^^ not found in this scope

error[E0412]: cannot find type `Arc` in this scope
  --> /home/user/.rustup/toolchains/nightly-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/src/libstd/io/lazy.rs:50:38
   |
50 |     unsafe fn init(&'static self) -> Arc<T> {
   |                                      ^^^ not found in this scope

error[E0412]: cannot find type `Arc` in this scope
   --> /home/user/.rustup/toolchains/nightly-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/src/libstd/io/stdio.rs:148:12
    |
148 |     inner: Arc<Mutex<BufReader<Maybe<StdinRaw>>>>,
    |            ^^^ not found in this scope

error[E0412]: cannot find type `Arc` in this scope
   --> /home/user/.rustup/toolchains/nightly-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/src/libstd/io/stdio.rs:207:24
    |
207 |     fn stdin_init() -> Arc<Mutex<BufReader<Maybe<StdinRaw>>>> {
    |                        ^^^ not found in this scope

error[E0433]: failed to resolve. Use of undeclared type or module `Arc`
   --> /home/user/.rustup/toolchains/nightly-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/src/libstd/io/stdio.rs:213:9
    |
213 |         Arc::new(Mutex::new(BufReader::with_capacity(stdio::STDIN_BUF_SIZE, stdin)))
    |         ^^^^^^^^ Use of undeclared type or module `Arc`

error[E0412]: cannot find type `Arc` in this scope
   --> /home/user/.rustup/toolchains/nightly-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/src/libstd/io/stdio.rs:347:12
    |
347 |     inner: Arc<ReentrantMutex<RefCell<LineWriter<Maybe<StdoutRaw>>>>>,
    |            ^^^ not found in this scope

error[E0412]: cannot find type `Arc` in this scope
   --> /home/user/.rustup/toolchains/nightly-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/src/libstd/io/stdio.rs:406:25
    |
406 |     fn stdout_init() -> Arc<ReentrantMutex<RefCell<LineWriter<Maybe<StdoutRaw>>>>> {
    |                         ^^^ not found in this scope

error[E0433]: failed to resolve. Use of undeclared type or module `Arc`
   --> /home/user/.rustup/toolchains/nightly-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/src/libstd/io/stdio.rs:411:9
    |
411 |         Arc::new(ReentrantMutex::new(RefCell::new(LineWriter::new(stdout))))
    |         ^^^^^^^^ Use of undeclared type or module `Arc`

error[E0412]: cannot find type `Arc` in this scope
   --> /home/user/.rustup/toolchains/nightly-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/src/libstd/io/stdio.rs:488:12
    |
488 |     inner: Arc<ReentrantMutex<RefCell<Maybe<StderrRaw>>>>,
    |            ^^^ not found in this scope

error[E0412]: cannot find type `Arc` in this scope
   --> /home/user/.rustup/toolchains/nightly-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/src/libstd/io/stdio.rs:541:25
    |
541 |     fn stderr_init() -> Arc<ReentrantMutex<RefCell<Maybe<StderrRaw>>>> {
    |                         ^^^ not found in this scope

error[E0433]: failed to resolve. Use of undeclared type or module `Arc`
   --> /home/user/.rustup/toolchains/nightly-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/src/libstd/io/stdio.rs:546:9
    |
546 |         Arc::new(ReentrantMutex::new(RefCell::new(stderr)))
    |         ^^^^^^^^ Use of undeclared type or module `Arc`

error[E0412]: cannot find type `Arc` in this scope
   --> /home/user/.rustup/toolchains/nightly-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/src/libstd/panic.rs:217:48
    |
217 | impl<T: RefUnwindSafe + ?Sized> UnwindSafe for Arc<T> {}
    |                                                ^^^ not found in this scope

error[E0412]: cannot find type `Arc` in this scope
  --> /home/user/.rustup/toolchains/nightly-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/src/libstd/sync/mpsc/blocking.rs:29:12
   |
29 |     inner: Arc<Inner>,
   |            ^^^ not found in this scope

error[E0412]: cannot find type `Arc` in this scope
  --> /home/user/.rustup/toolchains/nightly-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/src/libstd/sync/mpsc/blocking.rs:33:12
   |
33 |     inner: Arc<Inner>,
   |            ^^^ not found in this scope

error[E0433]: failed to resolve. Use of undeclared type or module `Arc`
  --> /home/user/.rustup/toolchains/nightly-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/src/libstd/sync/mpsc/blocking.rs:41:17
   |
41 |     let inner = Arc::new(Inner {
   |                 ^^^^^^^^ Use of undeclared type or module `Arc`

error[E0412]: cannot find type `Arc` in this scope
   --> /home/user/.rustup/toolchains/nightly-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/src/libstd/sync/mpsc/mod.rs:550:12
    |
550 |     inner: Arc<sync::Packet<T>>,
    |            ^^^ not found in this scope

error[E0412]: cannot find type `Arc` in this scope
   --> /home/user/.rustup/toolchains/nightly-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/src/libstd/sync/mpsc/mod.rs:652:13
    |
652 |     Oneshot(Arc<oneshot::Packet<T>>),
    |             ^^^ not found in this scope

error[E0412]: cannot find type `Arc` in this scope
   --> /home/user/.rustup/toolchains/nightly-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/src/libstd/sync/mpsc/mod.rs:653:12
    |
653 |     Stream(Arc<stream::Packet<T>>),
    |            ^^^ not found in this scope

error[E0412]: cannot find type `Arc` in this scope
   --> /home/user/.rustup/toolchains/nightly-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/src/libstd/sync/mpsc/mod.rs:654:12
    |
654 |     Shared(Arc<shared::Packet<T>>),
    |            ^^^ not found in this scope

error[E0412]: cannot find type `Arc` in this scope
   --> /home/user/.rustup/toolchains/nightly-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/src/libstd/sync/mpsc/mod.rs:655:10
    |
655 |     Sync(Arc<sync::Packet<T>>),
    |          ^^^ not found in this scope

error[E0433]: failed to resolve. Use of undeclared type or module `Arc`
   --> /home/user/.rustup/toolchains/nightly-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/src/libstd/sync/mpsc/mod.rs:723:13
    |
723 |     let a = Arc::new(oneshot::Packet::new());
    |             ^^^^^^^^ Use of undeclared type or module `Arc`

error[E0433]: failed to resolve. Use of undeclared type or module `Arc`
   --> /home/user/.rustup/toolchains/nightly-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/src/libstd/sync/mpsc/mod.rs:776:13
    |
776 |     let a = Arc::new(sync::Packet::new(bound));
    |             ^^^^^^^^ Use of undeclared type or module `Arc`

error[E0433]: failed to resolve. Use of undeclared type or module `Arc`
   --> /home/user/.rustup/toolchains/nightly-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/src/libstd/sync/mpsc/mod.rs:828:29
    |
828 |                     let a = Arc::new(stream::Packet::new());
    |                             ^^^^^^^^ Use of undeclared type or module `Arc`

error[E0433]: failed to resolve. Use of undeclared type or module `Arc`
   --> /home/user/.rustup/toolchains/nightly-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/src/libstd/sync/mpsc/mod.rs:865:25
    |
865 |                 let a = Arc::new(shared::Packet::new());
    |                         ^^^^^^^^ Use of undeclared type or module `Arc`

error[E0433]: failed to resolve. Use of undeclared type or module `Arc`
   --> /home/user/.rustup/toolchains/nightly-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/src/libstd/sync/mpsc/mod.rs:879:25
    |
879 |                 let a = Arc::new(shared::Packet::new());
    |                         ^^^^^^^^ Use of undeclared type or module `Arc`

error[E0412]: cannot find type `Arc` in this scope
   --> /home/user/.rustup/toolchains/nightly-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/src/libstd/sync/mpsc/mod.rs:931:19
    |
931 |     fn new(inner: Arc<sync::Packet<T>>) -> SyncSender<T> {
    |                   ^^^ not found in this scope

error[E0433]: failed to resolve. Could not find `AtomicUsize` in `atomic`
   --> /home/user/.rustup/toolchains/nightly-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/src/libstd/sys_common/thread_local.rs:130:18
    |
130 |             key: atomic::AtomicUsize::new(0),
    |                  ^^^^^^^^^^^^^^^^^^^^^^^^ Could not find `AtomicUsize` in `atomic`

error[E0412]: cannot find type `AtomicUsize` in module `atomic`
  --> /home/user/.rustup/toolchains/nightly-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/src/libstd/sys_common/util.rs:19:25
   |
19 |     static MIN: atomic::AtomicUsize = atomic::AtomicUsize::new(0);
   |                         ^^^^^^^^^^^ not found in `atomic`

error[E0433]: failed to resolve. Could not find `AtomicUsize` in `atomic`
  --> /home/user/.rustup/toolchains/nightly-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/src/libstd/sys_common/util.rs:19:39
   |
19 |     static MIN: atomic::AtomicUsize = atomic::AtomicUsize::new(0);
   |                                       ^^^^^^^^^^^^^^^^^^^^^^^^ Could not find `AtomicUsize` in `atomic`

error[E0412]: cannot find type `Arc` in this scope
  --> /home/user/.rustup/toolchains/nightly-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/src/libstd/sys/unix/fs.rs:51:11
   |
51 |     root: Arc<PathBuf>,
   |           ^^^ not found in this scope

error[E0412]: cannot find type `Arc` in this scope
  --> /home/user/.rustup/toolchains/nightly-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/src/libstd/sys/unix/fs.rs:61:11
   |
61 |     root: Arc<PathBuf>,
   |           ^^^ not found in this scope

error[E0433]: failed to resolve. Use of undeclared type or module `Arc`
   --> /home/user/.rustup/toolchains/nightly-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/src/libstd/sys/unix/fs.rs:627:16
    |
627 |     let root = Arc::new(p.to_path_buf());
    |                ^^^^^^^^ Use of undeclared type or module `Arc`

error[E0207]: the type parameter `T` is not constrained by the impl trait, self type, or predicates
   --> /home/user/.rustup/toolchains/nightly-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/src/libstd/panic.rs:217:6
    |
217 | impl<T: RefUnwindSafe + ?Sized> UnwindSafe for Arc<T> {}
    |      ^ unconstrained type parameter

error: aborting due to 66 previous errors

error: Could not compile `std`.

To learn more, run the command again with --verbose.
error: `"cargo" "build" "--release" "--manifest-path" "/tmp/xargo.d7ZAroGD4EqZ/Cargo.toml" "--target" "armv5te-unknown-linux-gnueabi" "-p" "std"` failed with exit code: Some(101)
note: run with `RUST_BACKTRACE=1` for a backtrace
</details>

[Related discussion](https://www.reddit.com/r/rust/comments/64j39d/crosscompiling_for_arm926ejs/)

I ask you kindly to help me solving this issue if it's possible to use rust std lib on armv5te linux.

**Comment by mosolovsa** [comment]
[Got answer here](https://github.com/joerg-krause/rust-cross-libs/issues/6)
TLDR: rustc target armv5te-unknown-linux-gnueabi shadowing custom target
